docs/function-format.md

```diff
@@ -58,42 +62,70 @@ The header keys and values would be populated with information about the functio
 
 Pros:
 
-* Streamable
+* Supports streaming
 * Common format
 
 Cons:
 
 * Requires a parsing library or fair amount of code to parse headers properly
 * Double parsing - headers + body (if body is to be parsed, such as json)
 
-#### JSON I/O Format (not implemented)
+#### JSON Input Format
 
 `--format json`
 
-The idea here is to keep the HTTP base structure, but make it a bit easier to parse by making the `request line` and `headers` a JSON struct.
-Eg:
+Fn accepts request data of the following format:
 
+```json
+{
+  "body": {
+    "some": "input"
+  }
+}
 ```
+
+Internally function receives data in following format:
+
+```json
 {
-  "request_url":"http://....",
-  "params": {
-    "blog_name": "yeezy"
+  "body": {
```

Comment:
still incorrect here, we shove request data into body strong, so `"body"` will be in the value here

Reply:
With the last commit it's right.

Here's what DumpRequest returns:
```
	json_test.go:57: {"body":"{\"a\":\"a\"}\n"
		,"headers":{"Content-Type":["application/json"],"Host":["localhost:8080"],"User-Agent":["curl/7.51.0"]}
		}

```

Reply:
your test is different than what you have in the docs. if you want to test that, you need to encode a struct like this:

```go
x := struct{
   Body struct{
       A string `json:"a"`
   } `json:"body"`
}{
  Body: struct{
    A: "a"
  }
}
```

so, the example in the docs is still incorrect. please address